pkg/ingester/ingester.go

```diff
@@ -1515,13 +1608,19 @@ func (i *Ingester) queryStreamSamples(ctx context.Context, db *userTSDB, from, t
 
 		it = series.Iterator(it)
 		for valType := it.Next(); valType != chunkenc.ValNone; valType = it.Next() {
-			if valType != chunkenc.ValFloat {
+			switch valType {
+			case chunkenc.ValFloat:
+				t, v := it.At()
+				ts.Samples = append(ts.Samples, mimirpb.Sample{Value: v, TimestampMs: t})
+			case chunkenc.ValHistogram:
+				// ignore
+			case chunkenc.ValFloatHistogram:
+				// ignore
+			default:
 				return 0, 0, fmt.Errorf("unsupported value type: %v", valType)
 			}
-			t, v := it.At()
-			ts.Samples = append(ts.Samples, mimirpb.Sample{Value: v, TimestampMs: t})
 		}
-		numSamples += len(ts.Samples)
+		numSamples += len(ts.Samples) + len(ts.Histograms)
```

Comment:
`ts.Histograms` will be always empty in this PR, right? (Just double-checking my understanding, nothing to change here.)

Reply:
yes, on the full branch `ts.Histograms` is loaded with samples, but here it's not https://github.com/grafana/mimir/blob/1e380d99e2bf28c7247309baa54ab732b8db6395/pkg/ingester/ingester.go#L1617